src/streams/EditStreamCard.js

```diff
@@ -5,45 +5,96 @@ import { View } from 'react-native';
 
 import Input from '../common/Input';
 import ZulipTextIntl from '../common/ZulipTextIntl';
-import SwitchRow from '../common/SwitchRow';
 import ZulipButton from '../common/ZulipButton';
-import styles, { createStyleSheet } from '../styles';
-import { IconPrivate } from '../common/Icons';
+import styles from '../styles';
+import SelectableOptionRow from '../common/SelectableOptionRow';
 
-const componentStyles = createStyleSheet({
-  switchRow: {
-    paddingLeft: 8,
-    paddingRight: 8,
-  },
-});
+type AccessPolicy =
+  | 'public'
+  | 'web_public'
+  | 'private_shared_history'
+  | 'private_protected_history';
+
+type StreamPolicySettings = {
+  // The properties on a stream object that are related to the access policy.
+  isPrivate: boolean,
+  isWebPublic: boolean,
+  historyPublicToSubscribers: boolean,
+};
```

Comment:
In the interest of having fewer distinct names for the same thing, let's give these the same snake-case names as in the API: `invite_only`, `is_web_public`, etc. Same for where they get added to `updateExistingStream`.

(I know we have some existing instances in main of `isPrivate` for `invite_only` and `isWebPublic` for `is_web_public` -- and I guess also an `inviteOnly` for `invite_only`. As there gets to be more of this logic and not just a single "private" / "public" boolean, I think that rapidly becomes less appealing. OK to leave those existing uses as is for this PR, but also if you feel like converting them over, that'd be a good additional commit or set of commits.)